### PR TITLE
fix(Scripts): Midsummer Music Range

### PR DESCRIPTION
**Pull request by ZhengPeiRu21**
## Changes Proposed:
-  Fix Midsummer Music to only play for players in visibility of invisible music doodad

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/3678

## Tests Performed:

- Tested in game, music players near object, flied far away on same map and the music correctly stopped and changed to zone music

## How to Test the Changes:

1. .go gameobject 50555
2. Listen to music
3. Run far way
4. Music should stop

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by acidmanifesto — APPROVED** [review]

**Comment by Gultask** [comment]
Didn't quite work as expected for me. Darnassus worked fine but Thunder Bluff's music reached all the way to Mulgore. (.go gameob 50556)
While Stormwind had no music at all, probably since I couldn't find its Music Doodad.

**Comment by ZhengPeiRu21** [comment]
Thank you for testing! The music packet tells the music to continue looping while within visibility distance, but it does take some time to stop once moving away - the music is about 1 minute long. This is the same behavior used by other PlayDirectMusic gobjects and is consistent with TrinityCore. If this is still different than official server behavior, though, we can try to more improve it.

For Stormwind music missing, I think there are many Midsummer objects that are missing spawns including many music doodads, but this is a separate issue.

**Comment by Gultask** [comment]
Oh yes, it stops looping. I think Darnassus is different because of the teleport transition into the city.

It's all good, then 👍 
I'll see which camps are missing the music and make a report about it!

File: src/server/scripts/World/go_scripts.cpp [hunk 1]
```diff
@@ -821,19 +821,21 @@ class go_midsummer_music : public GameObjectScript
                             if (!IsHolidayActive(HOLIDAY_FIRE_FESTIVAL))
                                 break;
 
-                            Map::PlayerList const& players = me->GetMap()->GetPlayers();
-                            for (Map::PlayerList::const_iterator itr = players.begin(); itr != players.end(); ++itr)
+                            std::list<Player*> targets;
+                            Acore::AnyPlayerInObjectRangeCheck check(me, me->GetVisibilityRange(), false);
+                            Acore::PlayerListSearcherWithSharedVision<Acore::AnyPlayerInObjectRangeCheck> searcher(me, targets, check);
+                            Cell::VisitWorldObjects(me, searcher, me->GetVisibilityRange());
+                            for (std::list<Player*>::const_iterator iter = targets.begin(); iter != targets.end(); ++iter)
```
Comment (Nefertumm) [inline src/server/scripts/World/go_scripts.cpp:828]:
```suggestion
                            for (Player const* player : targets)
```

Maybe? :)

Comment (ZhengPeiRu21) [reply src/server/scripts/World/go_scripts.cpp:828]: That is much improvement! Haha. Thank you!

**Comment by Nefertumm** [comment]
Ah, well, remove the const, it is complaining

**Comment by acidmanifesto** [comment]
labeling to run build as the build work flow got stuck and is unable to be merged. will look it over when tests are done.

**Comment by acidmanifesto** [comment]
Thank you for the PR

**Pull request merged**